omod-1.8/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_8/TaskActionController1_8Test.java

```diff
@@ -0,0 +1,137 @@
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.webservices.rest.web.v1_0.controller.openmrs1_8;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.module.webservices.helper.TaskAction;
+import org.openmrs.module.webservices.rest.web.MockTaskServiceWrapper;
+import org.openmrs.module.webservices.rest.web.api.RestService;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest;
+import org.openmrs.module.webservices.rest.web.v1_0.resource.openmrs1_8.TaskActionResource1_8;
+import org.openmrs.module.webservices.rest.web.v1_0.resource.openmrs1_8.TaskDefinitionResource1_8;
+import org.openmrs.scheduler.TaskDefinition;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import static org.hamcrest.CoreMatchers.hasItem;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.IsNot.not;
+
+public class TaskActionController1_8Test extends MainResourceControllerTest {
+	
+	@Autowired
+	RestService restService;
+	
+	private TaskDefinition testTask = new TaskDefinition(1, "TestTask", "TestTask Description",
+	        "org.openmrs.scheduler.tasks.TestTask");
+	
+	private String testTaskMockJson = "{\"id\" : 1,\"name\":\"TestTask\",\"description\":\"TestTask Description\","
+	        + " \"taskClass\":\"org.openmrs.scheduler.tasks.TestTask\"}";
+	
+	private MockTaskServiceWrapper mockTaskServiceWrapper = new MockTaskServiceWrapper();
+	
+	@Before
+	public void setUp() throws Exception {
+		mockTaskServiceWrapper.registeredTasks.add(testTask.getName());
+		
+		TaskActionResource1_8 taskActionResource = (TaskActionResource1_8) restService
+		        .getResourceBySupportedClass(TaskAction.class);
+		taskActionResource.setTaskServiceWrapper(mockTaskServiceWrapper);
+		
+		TaskDefinitionResource1_8 taskResource = (TaskDefinitionResource1_8) restService
+		        .getResourceBySupportedClass(TaskDefinition.class);
+		taskResource.setTaskServiceWrapper(mockTaskServiceWrapper);
+	}
+	
+	@Test
+	public void shouldStartTestTask() throws Exception {
+		//sanity check
+		assertThat(mockTaskServiceWrapper.scheduledTasks, not(hasItem(testTask.getName())));
+		deserialize(handle(newPostRequest(getURI(), "{\"action\": \"scheduletask\", \"tasks\":[" + testTaskMockJson + "]}")));
+		assertThat(mockTaskServiceWrapper.scheduledTasks, hasItem(testTask.getName()));
+	}
+	
+	@Test
+	public void shouldStopTestTask() throws Exception {
+		mockTaskServiceWrapper.scheduledTasks.add(testTask.getName());
+		deserialize(handle(newPostRequest(getURI(), "{\"action\": \"shutdowntask\", \"tasks\":[" + testTaskMockJson + "]}")));
+		assertThat(mockTaskServiceWrapper.scheduledTasks, not(hasItem(testTask.getName())));
+	}
+	
+	@Test
+	public void shouldDoNothingIfTestTaskAlreadyStarted() throws Exception {
```

Comment:
What should do nothing?

Reply:
That means, When I call "SheduleTask" action for the already scheduled task, It will not affect the status of that task.
Doing this test just for the confirmation.